tests/test_translator.py

```diff
@@ -0,0 +1,161 @@
+DEEPL_ENV_VAR = "DEEPL_API_KEY"
+
+
+@pytest.fixture
+def config():
+    config = _config
+    return config
+
+
+def test_split_input_text():
+    input_text = "Hello: How are you?\nI am fine: Thank you."
+    expected_output = ["Hello", " How are you?", "I am fine", " Thank you."]
+    assert split_input_text(input_text) == expected_output
+
+    input_text = "Hello\nHow are you?\nI am fine\nThank you."
+    expected_output = ["Hello", "How are you?", "I am fine", "Thank you."]
+    assert split_input_text(input_text) == expected_output
+
+
+@pytest.mark.parametrize("translator_class, lang_spec, model_name, tokenizer_name", [
+    (LocalTranslator, "ja", "facebook/m2m100_1.2B", "facebook/m2m100_1.2B"),
+    (LocalDanTranslator, "ja", "facebook/m2m100_1.2B", "facebook/m2m100_1.2B"),
+    (LocalEncodingTranslator, "ja", "facebook/m2m100_1.2B", "facebook/m2m100_1.2B"),
+    (LocalGoodsideTranslator, "ja", "facebook/m2m100_1.2B", "facebook/m2m100_1.2B"),
+    (LocalTranslator, "fr", "facebook/m2m100_1.2B", "facebook/m2m100_1.2B"),
+    (LocalTranslator, "jap", "Helsinki-NLP/opus-mt-en-{}", "Helsinki-NLP/opus-mt-en-{}"),
+    (LocalDanTranslator, "jap", "Helsinki-NLP/opus-mt-en-{}", "Helsinki-NLP/opus-mt-en-{}"),
+    (LocalEncodingTranslator, "jap", "Helsinki-NLP/opus-mt-en-{}", "Helsinki-NLP/opus-mt-en-{}"),
+    (LocalGoodsideTranslator, "jap", "Helsinki-NLP/opus-mt-en-{}", "Helsinki-NLP/opus-mt-en-{}"),
+    (LocalTranslator, "fr", "Helsinki-NLP/opus-mt-en-{}", "Helsinki-NLP/opus-mt-en-{}"),
+])
+def test_local_translate_single_language(translator_class, lang_spec, model_name, tokenizer_name):
+    config = _config
+    config.run.local_model_name = model_name 
+    config.run.local_tokenizer_name = tokenizer_name 
+    config.run.lang_spec = lang_spec
+    translator = translator_class(config_root=config)
+    input_text = "Hello, how are you?"
+    
+    translated_text = translator._get_response(input_text, 
+        source_lang="en", 
+        target_lang=config.run.lang_spec)
+    assert isinstance(translated_text, str)
+    assert translated_text != input_text
+
+
+@pytest.mark.parametrize("translator_class, lang_specs, model_name, tokenizer_name", [
+    (LocalTranslator, ["ja", "fr"], "facebook/m2m100_1.2B", "facebook/m2m100_1.2B"),
+    (LocalDanTranslator, ["ja", "fr"], "facebook/m2m100_1.2B", "facebook/m2m100_1.2B"),
+    (LocalEncodingTranslator, ["ja", "fr"], "facebook/m2m100_1.2B", "facebook/m2m100_1.2B"),
+    (LocalGoodsideTranslator, ["ja", "fr"], "facebook/m2m100_1.2B", "facebook/m2m100_1.2B"),
+    (LocalTranslator, ["jap", "fr"], "Helsinki-NLP/opus-mt-en-{}", "Helsinki-NLP/opus-mt-en-{}"),
+    (LocalDanTranslator, ["jap", "fr"], "Helsinki-NLP/opus-mt-en-{}", "Helsinki-NLP/opus-mt-en-{}"),
+    (LocalEncodingTranslator, ["jap", "fr"], "Helsinki-NLP/opus-mt-en-{}", "Helsinki-NLP/opus-mt-en-{}"),
+    (LocalGoodsideTranslator, ["jap", "fr"], "Helsinki-NLP/opus-mt-en-{}", "Helsinki-NLP/opus-mt-en-{}"),
+])
```

Comment:
Testing in `github` actions does not have access to GPU enabled systems, also this project cannot expect a GPU to be available in default environments.

Reply:
I try to use `CPU` default value for `self.device` in the `translator.py`.